File: src/main/java/de/epages/WebServices/AssignmentService/AssignmentServiceClient.java
package de.epages.WebServices.AssignmentService;

import java.rmi.RemoteException;
import java.util.ArrayList;
import java.util.List;
import java.util.logging.Logger;

import de.epages.WebServices.WebServiceConfiguration;
import de.epages.WebServices.AssignmentService.Stub.AssignmentService;
import de.epages.WebServices.AssignmentService.Stub.AssignmentServiceLocator;
import de.epages.WebServices.AssignmentService.Stub.Bind_Assignment_SOAPStub;
import de.epages.WebServices.AssignmentService.Stub.TAssignProductToCatalog_Input;
import de.epages.WebServices.AssignmentService.Stub.TAssignProductToCatalog_Return;
import de.epages.WebServices.AssignmentService.Stub.TGetCatalogsAssignedToProduct_Return;
import de.epages.WebServices.AssignmentService.Stub.TGetProductsAssignedToCatalog_Return;
import de.epages.WebServices.AssignmentService.Stub.TIsProductAssignedToCatalog_Input;
import de.epages.WebServices.AssignmentService.Stub.TIsProductAssignedToCatalog_Return;
import de.epages.WebServices.AssignmentService.Stub.TRemoveProductFromCatalog_Input;
import de.epages.WebServices.AssignmentService.Stub.TRemoveProductFromCatalog_Return;

public class AssignmentServiceClient {
    private AssignmentService service;
    private Bind_Assignment_SOAPStub stub;
    private static Logger log = Logger.getLogger(AssignmentServiceClient.class.getName());

    /**
     * Class constructor
     */
    public AssignmentServiceClient(WebServiceConfiguration config) {
        service = new AssignmentServiceLocator();
        log.info("address specified by wsdl: " + service.getport_AssignmentAddress());
        log.info("using web service Url: " + config.getWebserviceURL());

        try {
            stub = new Bind_Assignment_SOAPStub(config.getWebserviceURL(), service);

            // setting user-path and password of the shop
            stub.setUsername(config.getUsername());
            stub.setPassword(config.getPassword());
        }
        catch (Exception e) {
            log.severe( e.toString() );
        }
    }

    /**
     * Gets a list of Catalogs assigned to a Product.
     * If there is an error with a single Assignment, then this Assignment will be skipped and
     * it will be not part of there result list.
     *
     * @param productPaths  list of productPaths
     * @return              list of TGetCatalogsAssignedToProduct_Return
     * @throws              RemoteException
     * @see                 TGetCatalogsAssignedToProduct_Return
     */
    public List<TGetCatalogsAssignedToProduct_Return> getCatalogsAssignedToProduct(String[] productPaths) throws RemoteException {
        log.info("getCatalogsAssignedToProduct called");
        TGetCatalogsAssignedToProduct_Return[] Assignments = stub.getCatalogsAssignedToProduct(productPaths);
        List<TGetCatalogsAssignedToProduct_Return> result = new ArrayList<TGetCatalogsAssignedToProduct_Return>();

        for (int i = 0; i < Assignments.length; i++) {
            TGetCatalogsAssignedToProduct_Return Assignment = Assignments[i];

            if (Assignment.getError() == null) {
                log.info("successfully retrieved Assignments to Product: " + Assignment.getProduct());
                result.add(Assignment);
            } else {
                log.severe("an error occured (Epages Error):\n" + Assignment.getError().getMessage());
            }
        }
        return result;
    }

    /**
     * Gets a list of Products assigned to a Catalog.
     * If there is an error with a single Assignment, then this Assignment will be skipped and
     * it will be not part of there result list.
     *
     * @param catalogPaths  list of catalogPaths
     * @return              list of TGetProductsAssignedToCatalog_Return
     * @throws              RemoteException
     * @see                 TGetProductsAssignedToCatalog_Return
     */
    public List<TGetProductsAssignedToCatalog_Return> getProductsAssignedToCatalog(String[] catalogPaths) throws RemoteException {
        log.info("getProductsAssignedToCatalog called");
        TGetProductsAssignedToCatalog_Return[] Assignments = stub.getProductsAssignedToCatalog(catalogPaths);
        List<TGetProductsAssignedToCatalog_Return> result = new ArrayList<TGetProductsAssignedToCatalog_Return>();

        for (int i = 0; i < Assignments.length; i++) {
            TGetProductsAssignedToCatalog_Return Assignment = Assignments[i];

            if (Assignment.getError() == null) {
                log.info("successfully retrieved Assignments to Catalog: " + Assignment.getCatalog());
                result.add(Assignment);
            } else {
                log.severe("an error occured (Epages Error):\n" + Assignment.getError().getMessage());
            }
        }
        return result;
    }

    /**
     * Assignes a product to a catalog
     * If there is an error with a single Assignment, then this Assignment will be skipped and
     * it will be not part of there result list.
     *
     * @param assignments list of TAssignProductToCatalog_Input
     * @return          list of TAssignProductToCatalog_Return
     * @throws          RemoteException
     * @see             TAssignProductToCatalog_Input
     * @see             TAssignProductToCatalog_Return
     */
    public List<TAssignProductToCatalog_Return> assignProductToCatalog(List<TAssignProductToCatalog_Input> assignments) throws RemoteException {
        log.info("assignProductToCatalog called");
        TAssignProductToCatalog_Input[] input = new TAssignProductToCatalog_Input[assignments.size()];
        for (int i = 0; i < assignments.size(); i++) {
            input[i] =  assignments.get(i);
        }

        TAssignProductToCatalog_Return[] results = stub.assignProductToCatalog(input);
        List<TAssignProductToCatalog_Return> resultList = new ArrayList<TAssignProductToCatalog_Return>();

        for (int i = 0; i < results.length; i++) {
            TAssignProductToCatalog_Return result = results[i];

            if (result.getError() == null) {
                log.info("successfully created Assignment: " + result.getCatalog() + "/" + result.getProduct());
                resultList.add(result);
            } else {
                log.severe("an error occured (Epages Error):\n" + result.getError().getMessage());
            }
        }

        return resultList;
    }

    /**
     * Removes an assignment of a Product and a Catalog.
     * If there is an error with a single Assignment, then this Assignment will be skipped and
     * it will be not part of there result list.
     *
     * @param assignments     list of assignments
     * @return          list of TRemoveProductFromCatalog_Return
     * @throws          RemoteException
     * @see             TRemoveProductFromCatalog_Return
     */
    public List<TRemoveProductFromCatalog_Return> removeProductFromCatalog(List<TRemoveProductFromCatalog_Input> assignments) throws RemoteException {
        log.info("removeProductFromCatalog called");
        TRemoveProductFromCatalog_Input[] input = new TRemoveProductFromCatalog_Input[assignments.size()];
        for (int i = 0; i < assignments.size(); i++) {
            input[i] =  assignments.get(i);
        }

        TRemoveProductFromCatalog_Return[] results = stub.removeProductFromCatalog(input);
        List<TRemoveProductFromCatalog_Return> resultList = new ArrayList<TRemoveProductFromCatalog_Return>();

        for (int i = 0; i < results.length; i++) {
            TRemoveProductFromCatalog_Return result = results[i];

            if (result.getError() == null) {
                log.info("successfully deleted Assignment: " + result.getCatalog() + "/" + result.getProduct());
                resultList.add(result);
            } else {
                log.severe("an error occured (Epages Error):\n" + result.getError().getMessage());
            }
        }

        return resultList;
    }

    /**
     * Checkes if a product is assigned to a catalog.
     * If there is an error with a single Assignment, then this Assignment will be skipped and
     * it will be not part of there result list.
     *
     * @param assignments list of Assignments
     * @return          list of TIsProductAssignedToCatalog_Return
     * @throws          RemoteException
     * @see             TIsProductAssignedToCatalog_Return
     */
    public List<TIsProductAssignedToCatalog_Return> isProductAssignedToCatalog(List<TIsProductAssignedToCatalog_Input> assignments) throws RemoteException {
        log.info("existsAssignment called");
        TIsProductAssignedToCatalog_Input[] input = new TIsProductAssignedToCatalog_Input[assignments.size()];
        for (int i = 0; i < assignments.size(); i++) {
            input[i] =  assignments.get(i);
        }

        TIsProductAssignedToCatalog_Return[] results = stub.isProductAssignedToCatalog(input);
        List<TIsProductAssignedToCatalog_Return> resultList = new ArrayList<TIsProductAssignedToCatalog_Return>();

        for (int i = 0; i < results.length; i++) {
            TIsProductAssignedToCatalog_Return result = results[i];

            if (result.getError() == null) {
                log.info("successfully check Assignment existance: " + result.getCatalog() + "/" + result.getProduct());
                resultList.add(result);
            } else {
                log.severe("an error occured (Epages Error):\n" + result.getError().getMessage());
            }
        }

        return resultList;
    }
}

File: dotnet/shop/EpagesWebServices/AssignmentServiceClient.cs
using System;
using EpagesWebServices.AssignmentServiceRef;
using System.Collections;

namespace EpagesWebServices
{
	/// <summary>
	/// Client that uses generated stub to call Assignment web services methods
	/// </summary>
	public class AssignmentServiceClient : WebServiceConfiguration
	{
		AssignmentService stub;

		public AssignmentServiceClient()
		{
			stub = new AssignmentService();
			stub.Credentials = new System.Net.NetworkCredential(WEBSERVICE_LOGIN, WEBSERVICE_PASSWORD);
			stub.PreAuthenticate = true;
			stub.Url = WEBSERVICE_URL;
		}

		/// <summary>
		/// use stub to assignment a Product to a Catalog via web services
		/// </summary>
		/// <param name="Assignments">Array of TAssignProductToCatalog_Input</param>
		/// <returns>ArrayList of TAssignProductToCatalog_Return</returns>
		public ArrayList assignProductToCatalog(TAssignProductToCatalog_Input[] Assignments) 
		{
			TAssignProductToCatalog_Return[] Assignments_out = stub.assignProductToCatalog(Assignments);

			ArrayList result = new ArrayList();

			for(int i = 0; i < Assignments_out.Length; i++)
			{
				TAssignProductToCatalog_Return Assignment_out = Assignments_out[i];

				if (Assignment_out.Error == null) 
				{
					Console.WriteLine("successfully assigned: " + Assignment_out.Catalog + "/" + Assignment_out.Product);
					result.Add(Assignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + Assignment_out.Error.Message);
				}
			}

			return result;
		}
		
		/// <summary>
		/// use stub to get Catalogs assigned to a Product via web services
		/// </summary>
		/// <param name="productPaths">products, Array of paths</param>
		/// <returns>ArrayList of TGetCatalogsAssignedToProduct_Return</returns>
		public ArrayList getCatalogsAssignedToProduct(string[] productPaths) 
		{
			TGetCatalogsAssignedToProduct_Return[] Assignments_out = stub.getCatalogsAssignedToProduct(productPaths);

			ArrayList result = new ArrayList();

			for(int i = 0; i < Assignments_out.Length; i++)
			{
				TGetCatalogsAssignedToProduct_Return Assignment_out = Assignments_out[i];

				if (Assignment_out.Error == null) 
				{
					Console.WriteLine("successfully retrieved Assignments for Product: " + Assignment_out.Product);
					result.Add(Assignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + Assignment_out.Error.Message);
				}
			}

			return result;
		}
		
		/// <summary>
		/// use stub to get Products assigned to a Catalog via web services
		/// </summary>
		/// <param name="catalogPaths">catalogs, Array of paths</param>
		/// <returns>ArrayList of TGetProductsAssignedToCatalog_Return</returns>
		public ArrayList getProductsAssignedToCatalog(string[] catalogPaths) 
		{
			TGetProductsAssignedToCatalog_Return[] Assignments_out = stub.getProductsAssignedToCatalog(catalogPaths);

			ArrayList result = new ArrayList();

			for(int i = 0; i < Assignments_out.Length; i++)
			{
				TGetProductsAssignedToCatalog_Return Assignment_out = Assignments_out[i];

				if (Assignment_out.Error == null) 
				{
					Console.WriteLine("successfully retrieved Assignments for Catalog: " + Assignment_out.Catalog);
					result.Add(Assignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + Assignment_out.Error.Message);
				}
			}

			return result;
		}
		
		/// <summary>
		/// use stub to remove an Assignment via web services
		/// </summary>
		/// <param name="assignments">Assignments to delete, Array of TRemoveProductFromCatalog_Input</param>
		/// <returns>ArrayList of TRemoveProductFromCatalog_Return</returns>
		public ArrayList removeProductFromCatalog(TRemoveProductFromCatalog_Input[] assignments) 
		{
			TRemoveProductFromCatalog_Return[] Assignments_out = stub.removeProductFromCatalog(assignments);

			ArrayList result = new ArrayList();

			for(int i = 0; i < Assignments_out.Length; i++)
			{
				TRemoveProductFromCatalog_Return Assignment_out = Assignments_out[i];

				if (Assignment_out.Error == null) 
				{
					Console.WriteLine("successfully removed Assignment: " + Assignment_out.Catalog + "/" + Assignment_out.Product);
					result.Add(Assignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + Assignment_out.Error.Message);
				}
			}

			return result;
		}

		/// <summary>
		/// use stub to check via web services if a Product and a Catalog are assigned
		/// </summary>
		/// <param name="assignments">Assignments to check, Array of TIsProductAssignedToCatalog_Input</param>
		/// <returns>ArrayList of TIsProductAssignedToCatalog_Return</returns>
		public ArrayList isProductAssignedToCatalog(TIsProductAssignedToCatalog_Input[] assignments) 
		{
			TIsProductAssignedToCatalog_Return[] Assignments_out = stub.isProductAssignedToCatalog(assignments);

			ArrayList result = new ArrayList();

			for(int i = 0; i < Assignments_out.Length; i++)
			{
				TIsProductAssignedToCatalog_Return Assignment_out = Assignments_out[i];

				if (Assignment_out.Error == null) 
				{
					Console.WriteLine("successfully checked Assignment: " + Assignment_out.Catalog + "/" + Assignment_out.Product);
					result.Add(Assignment_out);
				} 
				else 
				{
					Console.WriteLine("an error occured (Epages Error):\n" + Assignment_out.Error.Message);
				}
			}

			return result;
		}
	}
}
